pkg/logging/formatter.go

```diff
@@ -0,0 +1,164 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"golang.org/x/exp/slog"
+)
+
+type formatter struct {
+	l *slog.Logger
+
+	exiter   exiter
+	panicker panicker
+}
+
+type exiter interface {
+	Exit(code int)
+}
+
+type defaultExiter struct{}
+
+func (de defaultExiter) Exit(code int) {
+	os.Exit(code)
+}
+
+type panicker interface {
+	Panic(msg string)
+}
+
+type defaultPanicker struct{}
+
+func (dp defaultPanicker) Panic(msg string) {
+	panic(msg)
+}
+
```

Comment:
What about `Debug()`, `Info()`, `Warn()` and `Error()` (non-formatting versions) to "complete" the formatter?